src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const answer: GroupsMap<T> = Object.fromEntries(
+    items.map((item) => [item[key] as string, []]),
```

Comment:
The use of `Object.fromEntries` combined with `map` here will not work as intended for grouping items by key. If there are duplicate keys in the `items` array, the subsequent keys will overwrite the previous ones, resulting in only the last item for each key being present in the final `answer` object. Instead, you should initialize the `answer` object as an empty object and then populate it within the loop by checking if the key exists and pushing to the array or creating a new array if it doesn't.